app/actions/index.js

```diff
@@ -258,11 +258,24 @@ export const updateTemporaryTitleValue = title => ({
   title
 })
 
-export const updateTitle = (key, path, editValue) => async dispatch => {
+export const updateTitle = (key, path, editValue) => async (dispatch, getState) => {
   const filePath = `${path}/dat.json`
-  const blob = await readFile(filePath)
-  const metadata = { ...JSON.parse(blob), title: editValue }
-  await writeFile(filePath, JSON.stringify(metadata))
+  let metadata
+
+  try {
+    metadata = await readFile(filePath)
+    metadata = JSON.parse(blob)
+  } catch (_) {
+    metadata = getState().dats[key].metadata
```

Comment:
I think its ambiguous to use either the file or the metadata, shouldn't this be just the metadata?

Reply:
Also: Theoretically the dat doesn't need to exist. Would be good to have an error here if it doesnt

Reply:
mhh, I though as well at first but this action is only used when changing a dat's title in the view so the assumption seemed alright!